src/video_core/renderer_opengl/gl_shader_decompiler.cpp

```diff
@@ -307,20 +322,18 @@ class GLSLGenerator {
     /// Generates code representing a source register.
     std::string GetSourceRegister(const SourceRegister& source_reg,
                                   u32 address_register_index) const {
-        u32 index = static_cast<u32>(source_reg.GetIndex());
-        std::string index_str = std::to_string(index);
+        const u32 index = static_cast<u32>(source_reg.GetIndex());
 
         switch (source_reg.GetRegisterType()) {
         case RegisterType::Input:
             return inputreg_getter(index);
         case RegisterType::Temporary:
-            return "reg_tmp" + index_str;
+            return fmt::format("reg_tmp{}", index);
         case RegisterType::FloatUniform:
             if (address_register_index != 0) {
-                index_str +=
-                    std::string(" + address_registers.") + "xyz"[address_register_index - 1];
+                return fmt::format("{} + address_registers.{}", index, "xyz"[address_register_index - 1]);
```

Comment:
fixes https://github.com/citra-emu/citra/issues/5334, https://github.com/citra-emu/citra/pull/5262#issuecomment-623763760, and https://github.com/citra-emu/citra/pull/5262#issuecomment-623796177
```suggestion
                return fmt::format("uniforms.f[{} + address_registers.{}]", index,
                                   "xyz"[address_register_index - 1]);
```